File: PHP/MajorityElementTest.php
<?php
/**
 * @version PHP 7.3.29
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit MajorityElementTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MajorityElement.php");

class MajorityElementTest extends TestCase {

    public function testMajorityElement() {
        $obj = new MajorityElement();
        //Test 1
        $want = 3;
        $got = $obj->majorityElement( [3,2,3] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 2;
        $got = $obj->majorityElement( [2,2,1,1,1,2,2] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/MajorityElement_test.go
/**
 * @version GO 1.19.4
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestMajorityElement
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMajorityElement(t *testing.T) {
	tables := []struct {
		want int
		nums []int
	}{
		{3, []int{3, 2, 3}},
		{2, []int{2, 2, 1, 1, 1, 2, 2}},
	}

	for _, table := range tables {
		got := majorityElement(table.nums)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
